lib/mapnik_backend.js

```diff
@@ -175,29 +166,21 @@ MapnikSource.prototype._open = function(uri) {
 
     // Initialize this map. This wraps `localize()` and calls `create()`
     // afterwards to actually create a new Mapnik map object.
-    Step(function() {
-        source._loadXML(this);
-    }, function(err, xml) {
-        if (err) return error(err);
-        // https://github.com/mapbox/tilelive-mapnik/issues/25
-        // there seems to be no value to assinging xml to a property
-        //source._xml = xml;
-        source._createMetatileCache(uri.query.metatileCache);
-        source._createPool(xml, this);
-    }, function(err) {
-        if (err) return error(err);
-        source._populateInfo(this);
-    }, function(err) {
-        if (err) return error(err);
-        source.open = true;
-        source.emit('open', null, source);
+    this._loadXML((err, xml) => {
+        if (err) return this.emit('open', err);
+
+        this._createMetatileCache(uri.query.metatileCache);
+        this._createPool(xml);
+        this._populateInfo((err_populate) => {
+            if (err_populate) return this.emit('open', err_populate);
+            this.open = true;
+            this.emit('open', null, this);
```

Comment:
Events don't work in the same way that callback to propagate errors. When an error happens you should emit the error event.

```
this.emit('error', err);
```
instead of
```
this.emit('open', err); // when error
this.emit('open', null, value); 
```

Reply:
Yeah, but it's setup so all parameters passed to `emit` are relied as-is to a callback function, so in this case they work like that. See https://github.com/Algunenano/tilelive-mapnik/blob/1458acbcbb9e8028adf1d030792f58f8a0ce48b2/lib/mapnik_backend.js#L33

We could change it to use 2 different events, but that's how it was set up previously (I didn't change anything there).